### PR TITLE
Add integration test for RunPodSandbox and CreateContainer

### PR DESCRIPTION
**Pull request by teawater**
Fixes: #1009

Signed-off-by: Hui Zhu <teawater@hyper.sh>

### CODE REVIEW

File: integration/container_create_test.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
+/*
+Copyright 2019 Authors.
```
Comment (mikebrow) [inline integration/container_create_test.go]: Copyright 2018 The containerd Authors.

File: integration/sandbox_run_test.go [hunk 2]
```diff
@@ -0,0 +1,39 @@
+/*
+Copyright 2019 Authors.
```
Comment (mikebrow) [inline integration/sandbox_run_test.go]: same

File: integration/sandbox_run_test.go [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	runtime "k8s.io/kubernetes/pkg/kubelet/apis/cri/runtime/v1alpha2"
+)
+
+func TestRunPodSandbox(t *testing.T) {
```
Comment (mikebrow) [inline integration/sandbox_run_test.go]: this is well tested over in cri tools and in the cri/integration/sandbox_clean_remove_test.go

File: integration/container_create_test.go [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+package integration
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	runtime "k8s.io/kubernetes/pkg/kubelet/apis/cri/runtime/v1alpha2"
+)
+
+func TestCreateContainer(t *testing.T) {
```
Comment (mikebrow) [inline integration/container_create_test.go]: This is already tested not needed here in other integration tests

File: integration/container_create_test.go [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+	sb, err := runtimeService.RunPodSandbox(sbConfig, *runtimeHandler)
+	require.NoError(t, err)
+	defer func() {
+		// Make sure the sandbox is cleaned up in any case.
+		runtimeService.StopPodSandbox(sb)
+		runtimeService.RemovePodSandbox(sb)
+	}()
+	config := &runtime.ContainerConfig{}
+	_, err = runtimeService.CreateContainer(sb, config, sbConfig)
+	require.EqualValues(t, "rpc error: code = Unknown desc = container config must include metadata", err.Error())
```
Comment (Random-Liu) [inline integration/container_create_test.go]: I think just assert an error is good enough, no need to match the error message.

Call runtimeServer.Status to make sure containerd is still running after this, this is to make sure containerd not panic.

File: integration/sandbox_run_test.go [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	runtime "k8s.io/kubernetes/pkg/kubelet/apis/cri/runtime/v1alpha2"
+)
+
+func TestRunPodSandboxWithoutMetadata(t *testing.T) {
+	sbConfig := &runtime.PodSandboxConfig{}
+	_, err := runtimeService.RunPodSandbox(sbConfig, *runtimeHandler)
+	require.EqualValues(t, "rpc error: code = Unknown desc = sandbox config must include metadata", err.Error())
```
Comment (Random-Liu) [inline integration/sandbox_run_test.go]: ditto.

**Comment by Random-Liu** [comment]
>  I'd just throw the two new tests into a /no_metadata_test.go?
agree.

**Comment by Random-Liu** [comment]
> I'd just throw the two new tests into a /no_metadata_test.go?

I still prefer this, will send a small PR to address.
@teawater Thanks for the PR!!

**Pull request merged**